leak_report.md

```diff
@@ -1,3 +1,2 @@
 # Leak report
-
-_Use this document to describe whatever memory leaks you find in `clean_whitespace.c` and how you might fix them. You should also probably remove this explanatory text._
+The reason that there is a memory leak is that the calloc used at line 41 is never freed. It is handed up until it makes it all the way to the isclean function, but then never gets freed. I believe that is where it should be freed. 
```

Comment:
```suggestion
The reason that there is a memory leak is that the calloc used at line 41 is never freed. It is handed up until it makes it all the way to the `is_clean` function, but then never gets freed. I believe that is where it should be freed. 
```

Does `valgrind` agree? Did that fix the `valgrind` errors? (Don't rely on "I think" statements when there are tools that can provide objective tests or measures.)